src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNIndexCache.java

```diff
@@ -314,14 +314,17 @@ public KNNIndexCacheEntry loadIndex(String indexPathUrl, String indexName) throw
         final WatcherHandle<FileWatcher> watcherHandle = resourceWatcherService.add(fileWatcher);
 
         // loadIndex from different library
-        if (KNNSettings.getKnnEngine(indexName).contains(KNNEngine.FAISS.getKnnEngineName())) {
-            final KNNFaissIndex knnIndex = KNNFaissIndex.loadIndex(indexPathUrl, getQueryParams(indexName), KNNSettings.getSpaceType(indexName));
-            return new KNNIndexCacheEntry(knnIndex, indexPathUrl, indexName, watcherHandle);
+        final KNNIndex knnIndex;
 
+        if (indexPathUrl.contains(KNNEngine.NMSLIB.getExtension())) {
```

Comment:
map be endsWith?

Reply:
So the reason I did contains instead of ends with is that the extension could be a compound extension. However, endswith is more correct, so I will fix